source/blog/2016-12-14-talking-to-docker-daemon.html.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Talk to Docker Daemon of Atomic Host remotely
+author: trishnag
+date: 2016-12-14 14:00:00 UTC
+published: true
+comments: true
+tags: atomic host, ansible, docker
+---
+
+
+This post will describe how to use [Docker](https://www.docker.com/) daemon of Atomic host remotely.  Note that we are also going to secure the Docker daemon since we are connecting via Network which we will be doing with TLS.
+
+TLS (Transport Layer Security) provides communication security over computer network. We will create client cert and server cert to secure our Docker daemon. OpenSSL will be used to to create the cert keys for establishing TLS connection.
+
+I am using [Fedora-Atomic](https://getfedora.org/en/atomic/) host as remote and [workstation](https://getfedora.org/en/workstation/download/) as my present host.
+
+Thanks to [Chris Houseknecht](https://twitter.com/CHouseknecht) for writing an [Ansible](https://www.ansible.com/) role which creates all the certs required automatically, so that there is no need to issue openssl commands manually. Here is the Ansible role repository: [role-secure-docker-daemon](https://github.com/ansible/role-secure-docker-daemon). Clone it to your present working host.
+
+```
+$ mkdir secure-docker-daemon
+$ cd secure-docker-daemon
+$ git clone https://github.com/ansible/role-secure-docker-daemon.git
+$ touch ansible.cfg inventory secure-docker-daemon.yml
+$ ls 
+ansible.cfg  inventory  role-secure-docker-daemon  secure-docker-daemon.yml
+```
+
+**ansible.cfg**:
+
+```
+$ vim ansible.cfg
+[defaults]
+inventory=inventory
+remote_user='USER_OF_ATOMIC_HOST'
+```
+
+**inventory**:
+
+```
+$ vim inventory 
+[serveratomic]
+'IP_OF_ATOMIC_HOST' ansible_ssh_private_key_file='PRIVATE_KEY_FILE'
+```
+
+**secure-docker-daemon.yml**:
+
+```
+$ vim secure-docker-daemon.yml
+---
+- name: Secure Docker daemon for Atomic host
+  hosts: serveratomic
+  gather_facts: no
+  become: yes
+  roles:
+    - role: role-secure-docker-daemon
+      dds_host: 'IP_OF_ATOMIC_HOST'
```

Comment:
it would be better to use {{ inventory_hostname }} here.  I'll add that to my version.